go/internal/initutil/datastore.go

```diff
@@ -86,14 +86,13 @@ func (m *Manager) getRootDatastore() (datastore.Batching, error) {
 		return sync_ds.MutexWrap(datastore.NewMapDatastore()), nil
 	}
 
-	var opts *badger.Options
-	if m.Datastore.FileIO {
-		opts = &badger.Options{
-			Options: badger.DefaultOptions.WithValueLogLoadingMode(badger_opts.FileIO),
-		}
+	opts := ipfsbadger.DefaultOptions
+	if m.Datastore.LowMemoryProfile {
+		applyBadgerLowMemoryProfile(m.initLogger, &opts)
+		opts.Options = ipfsbadger.DefaultOptions.WithValueLogLoadingMode(badger_opts.FileIO)
```

Comment:
isn't this line making the previous one useless?

anyway, I prefer that you move the 4 lines in `applyBadgerLowMemoryProfile` right here